📌 Module 1 — Product Performance Summary

Business Requirement

Create one table.

Expected output:

Product ID	Product Name	Category	Sub Category	Sales	Profit	Quantity	Profit Margin

You'll need to calculate:

Total Sales  
Total Profit  
Quantity Sold  
Profit Margin %

In [0]:
%sql
select `Product ID`, `Product Name`, `Category` , `Sub-Category`, round(sum(`Sales`),2) as Total_Sales, round(sum(`Profit`),2) as Total_Profit, sum(`Quantity`) as Total_Quantity, round((sum(`Profit`) / sum(`Sales`) )*100, 2) as Profit_Margin,
rank() over(order by sum(`Sales`) desc) as TotalSales_Rank,
rank() over(order by sum(`Profit`) desc) as TotalProfit_Rank,
rank() over(order by (sum(`Profit`) / sum(`Sales`) )*100 desc) as TotalMarginProfit_Rank
from abc_retail_pvt_ltd.data_analytics.superstore
group by `Product ID`, `Product Name`, `Category`, `Sub-Category`
ORDER BY Total_Sales DESC;

Product ID,Product Name,Category,Sub-Category,Total_Sales,Total_Profit,Total_Quantity,Profit_Margin,TotalSales_Rank,TotalProfit_Rank,TotalMarginProfit_Rank
TEC-CO-10004722,Canon imageCLASS 2200 Advanced Copier,Technology,Copiers,61599.82,25199.93,20,40.91,1,1,358
OFF-BI-10003527,Fellowes PB500 Electric Punch Plastic Comb Binding Machine with Manual Bind,Office Supplies,Binders,27453.38,7753.04,31,28.24,2,2,741
TEC-MA-10002412,Cisco TelePresence System EX90 Videoconferencing Unit,Technology,Machines,22638.48,-1811.08,6,-8.00,3,1888,1707
FUR-CH-10002024,HON 5400 Series Task Chairs for Big and Tall,Furniture,Chairs,21870.58,0.00,39,0.00,4,1585,1585
OFF-BI-10001359,GBC DocuBind TL300 Electric Binding System,Office Supplies,Binders,19823.48,2233.51,37,11.27,5,19,1337
OFF-BI-10000545,GBC Ibimaster 500 Manual ProClick Binding System,Office Supplies,Binders,19024.50,760.98,48,4.00,6,101,1517
TEC-CO-10001449,Hewlett Packard LaserJet 3310 Copier,Technology,Copiers,18839.69,6983.88,38,37.07,7,3,495
TEC-MA-10001127,"HP Designjet T520 Inkjet Large Format Printer - 24"" Color",Technology,Machines,18374.90,4094.98,12,22.29,8,5,964
OFF-BI-10004995,GBC DocuBind P400 Electric Binding System,Office Supplies,Binders,17965.07,-1878.17,27,-10.45,9,1889,1736
OFF-SU-10000151,High Speed Automatic Electric Letter Opener,Office Supplies,Supplies,17030.31,-262.00,11,-1.54,10,1821,1613


📌 Module 2 — Product Ranking

Rank products by

Revenue  
Profit  
Quantity Sold

In [0]:
%sql
SELECT `Product ID`, `Product Name`, ROUND(SUM(`Sales`),2) AS Total_Sales, ROUND(SUM(`Profit`),2) AS Total_Profit,
SUM(`Quantity`) AS Total_Quantity,
RANK() OVER(ORDER BY SUM(`Sales`) DESC) AS Sales_Rank,
RANK() OVER(ORDER BY SUM(`Profit`) DESC) AS Profit_Rank,
RANK() OVER(ORDER BY SUM(`Quantity`) DESC) AS Quantity_Rank
FROM abc_retail_pvt_ltd.data_analytics.superstore
GROUP BY `Product ID`, `Product Name`;

Product ID,Product Name,Total_Sales,Total_Profit,Total_Quantity,Sales_Rank,Profit_Rank,Quantity_Rank
TEC-CO-10004722,Canon imageCLASS 2200 Advanced Copier,61599.82,25199.93,20,1,1,833
OFF-BI-10003527,Fellowes PB500 Electric Punch Plastic Comb Binding Machine with Manual Bind,27453.38,7753.04,31,2,2,306
TEC-MA-10002412,Cisco TelePresence System EX90 Videoconferencing Unit,22638.48,-1811.08,6,3,1888,1721
FUR-CH-10002024,HON 5400 Series Task Chairs for Big and Tall,21870.58,0.00,39,4,1585,126
OFF-BI-10001359,GBC DocuBind TL300 Electric Binding System,19823.48,2233.51,37,5,19,156
OFF-BI-10000545,GBC Ibimaster 500 Manual ProClick Binding System,19024.50,760.98,48,6,101,39
TEC-CO-10001449,Hewlett Packard LaserJet 3310 Copier,18839.69,6983.88,38,7,3,143
TEC-MA-10001127,"HP Designjet T520 Inkjet Large Format Printer - 24"" Color",18374.90,4094.98,12,8,5,1410
OFF-BI-10004995,GBC DocuBind P400 Electric Binding System,17965.07,-1878.17,27,9,1889,449
OFF-SU-10000151,High Speed Automatic Electric Letter Opener,17030.31,-262.00,11,10,1821,1475


📌 Module 3 — Category Performance

Business asks

Which category contributes the most revenue?

Analyze

Furniture  
Office Supplies  
Technology

In [0]:
%sql
SELECT Category, ROUND(SUM(`Sales`), 2) AS Total_Sales,
ROUND(SUM(`Profit`),2) AS Total_Profit,
SUM(`Quantity`) AS Quantity,
ROUND((SUM(`Profit`)/SUM(`Sales`))*100,2) AS Profit_Margin,
rank() over(order by sum(`Sales`) desc) as TotalSales_Rank
FROM abc_retail_pvt_ltd.data_analytics.superstore
GROUP BY Category

Category,Total_Sales,Total_Profit,Quantity,Profit_Margin,TotalSales_Rank
Technology,839893.28,146543.38,7017,17.45,1
Furniture,754747.76,19730.00,8369,2.61,2
Office Supplies,731893.31,126023.44,23268,17.22,3


Finding

Technology generated the highest revenue of $839,893, making it the company's best-performing category.

Business Insight

Customers have higher demand for technology products compared to other categories.

Recommendation

Increase inventory and marketing investments for Technology while continuing to monitor profitability.

📌 Module 4 — Sub-Category Performance

Now drill deeper.

Instead of Category,

analyze

Chairs
Phones
Tables
Binders
Storage

etc.

In [0]:
%sql
SELECT`Sub-Category`, ROUND(SUM(`Sales`),2) AS Total_Sales, ROUND(SUM(`Profit`),2) AS Total_Profit, SUM(`Quantity`) AS Quantity,
ROUND((SUM(`Profit`)/SUM(`Sales`))*100,2) AS Profit_Margin,
rank() over(order by sum(`Sales`) desc) as TotalSales_Rank
FROM abc_retail_pvt_ltd.data_analytics.superstore
GROUP BY `Sub-Category`

Sub-Category,Total_Sales,Total_Profit,Quantity,Profit_Margin,TotalSales_Rank
Chairs,335768.25,27223.53,2431,8.11,1
Phones,331842.64,45050.83,3357,13.58,2
Storage,224644.55,21285.11,3191,9.48,3
Tables,208020.18,-17753.21,1261,-8.53,4
Binders,207354.88,31426.10,6071,15.16,5
Machines,189925.03,3461.98,442,1.82,6
Accessories,167380.32,41936.64,2976,25.05,7
Copiers,150745.29,56093.94,242,37.21,8
Bookcases,115361.20,-3632.07,878,-3.15,9
Appliances,108213.18,18329.48,1750,16.94,10


📌 Module 5 — High Revenue, Low Profit Products

One of my favorite analyses.

Business Question

Which products sell a lot but barely make money?

Why?

Maybe

discounts are too high
shipping costs are high
product cost is high

These products need investigation.

In [0]:
%sql
WITH product_performance AS 
(
SELECT `Product ID`, `Product Name`, ROUND(SUM(Sales), 2) AS Total_Sales, ROUND(SUM(Profit), 2) AS Total_Profit,
RANK() OVER (ORDER BY SUM(Sales) DESC) AS Sales_Rank,
RANK() OVER (ORDER BY SUM(Profit) ASC) AS Profit_Rank,
ROUND((SUM(`Profit`)/SUM(`Sales`))*100,2) AS Profit_Margin
FROM abc_retail_pvt_ltd.data_analytics.superstore
GROUP BY `Product ID`,`Product Name`
)

SELECT * FROM product_performance
WHERE Total_Sales>10000
ORDER BY Profit_Margin ASC;

Product ID,Product Name,Total_Sales,Total_Profit,Sales_Rank,Profit_Rank,Profit_Margin
TEC-MA-10000418,Cubify CubeX 3D Printer Double Head Print,11099.96,-8879.97,24,1,-80.00
TEC-MA-10000822,Lexmark MX611dhe Monochrome Laser Printer,16829.90,-4589.97,11,2,-27.27
OFF-BI-10004995,GBC DocuBind P400 Electric Binding System,17965.07,-1878.17,9,6,-10.45
TEC-MA-10002412,Cisco TelePresence System EX90 Videoconferencing Unit,22638.48,-1811.08,3,7,-8.00
OFF-SU-10002881,Martin Yale Chadless Opener Electric Letter Opener,16656.20,-1299.18,12,8,-7.80
FUR-BO-10004834,"Riverside Palais Royal Lawyers Bookcase, Royale Cherry Finish",15610.97,-669.54,14,25,-4.29
FUR-TA-10003473,Bretford Rectangular Conference Table Tops,13565.57,-359.35,17,60,-2.65
OFF-SU-10000151,High Speed Automatic Electric Letter Opener,17030.31,-262.00,10,74,-1.54
FUR-CH-10002024,HON 5400 Series Task Chairs for Big and Tall,21870.58,0.00,4,310,0.00
FUR-BO-10002213,DMI Eclipse Executive Suite Bookcases,11046.61,90.18,25,1257,0.82


📌 Module 6 — Loss Making Products

Business asks

Which products are losing money?

Expected output

Product	Sales	Profit

Sort by

Lowest Profit.

In [0]:
%sql
WITH loss_products AS 
(
SELECT `Product ID`, `Product Name`, ROUND(SUM(Sales), 2) AS Total_Sales, ROUND(SUM(Profit), 2) AS Total_Profit,
ROUND((SUM(`Profit`)/SUM(`Sales`))*100,2) AS Profit_Margin,
DENSE_RANK() OVER (ORDER BY SUM(Profit) ASC) AS Loss_Rank
FROM abc_retail_pvt_ltd.data_analytics.superstore
GROUP BY `Product ID`, `Product Name`
)

SELECT *
FROM loss_products
WHERE Total_Profit < 0
ORDER BY Loss_Rank;

Product ID,Product Name,Total_Sales,Total_Profit,Profit_Margin,Loss_Rank
TEC-MA-10000418,Cubify CubeX 3D Printer Double Head Print,11099.96,-8879.97,-80.00,1
TEC-MA-10000822,Lexmark MX611dhe Monochrome Laser Printer,16829.90,-4589.97,-27.27,2
TEC-MA-10004125,Cubify CubeX 3D Printer Triple Head Print,7999.98,-3839.99,-48.00,3
FUR-TA-10000198,Chromcraft Bull-Nose Wood Oval Conference Tables & Bases,9917.64,-2876.12,-29.00,4
FUR-TA-10001889,Bush Advantage Collection Racetrack Conference Table,9544.72,-1934.40,-20.27,5
OFF-BI-10004995,GBC DocuBind P400 Electric Binding System,17965.07,-1878.17,-10.45,6
TEC-MA-10002412,Cisco TelePresence System EX90 Videoconferencing Unit,22638.48,-1811.08,-8.00,7
OFF-SU-10002881,Martin Yale Chadless Opener Electric Letter Opener,16656.20,-1299.18,-7.80,8
FUR-TA-10001950,Balt Solid Wood Round Tables,6518.75,-1201.06,-18.42,9
FUR-TA-10004289,BoxOffice By Design Rectangular and Half-Moon Meeting Room Tables,1706.25,-1148.44,-67.31,10


📌 Module 7 — Product Segmentation

Similar to customers.

Example

Sales > 15000

↓

Best Seller

Sales between 8000-15000

↓

High Performer

Sales between 3000-8000

↓

Average Performer

Below 3000

↓

Low Performer

In [0]:
%sql
SELECT `Product Name`, ROUND(SUM(Sales), 2) AS Total_Sales,
CASE
WHEN SUM(Sales) > 15000 THEN 'Best Seller'
WHEN SUM(Sales) BETWEEN 8000 AND 15000 THEN 'High Performer'
WHEN SUM(Sales) BETWEEN 3000 AND 8000 THEN 'Average Performer'
ELSE 'Low Performer'
END AS Product_Segment
FROM abc_retail_pvt_ltd.data_analytics.superstore
GROUP BY `Product Name`
ORDER BY Total_Sales DESC;

Product Name,Total_Sales,Product_Segment
Canon imageCLASS 2200 Advanced Copier,61599.82,Best Seller
Fellowes PB500 Electric Punch Plastic Comb Binding Machine with Manual Bind,27453.38,Best Seller
Cisco TelePresence System EX90 Videoconferencing Unit,22638.48,Best Seller
HON 5400 Series Task Chairs for Big and Tall,21870.58,Best Seller
GBC DocuBind TL300 Electric Binding System,19823.48,Best Seller
GBC Ibimaster 500 Manual ProClick Binding System,19024.50,Best Seller
Hewlett Packard LaserJet 3310 Copier,18839.69,Best Seller
"HP Designjet T520 Inkjet Large Format Printer - 24"" Color",18374.90,Best Seller
GBC DocuBind P400 Electric Binding System,17965.07,Best Seller
High Speed Automatic Electric Letter Opener,17030.31,Best Seller


In [0]:
%sql
WITH product_segment AS 
(
SELECT `Product Name`, SUM(Sales) AS Total_Sales,
CASE
WHEN SUM(Sales) > 15000 THEN 'Best Seller'
WHEN SUM(Sales) BETWEEN 8000 AND 15000 THEN 'High Performer'
WHEN SUM(Sales) BETWEEN 3000 AND 8000 THEN 'Average Performer'
ELSE 'Low Performer'
END AS Product_Segment
FROM abc_retail_pvt_ltd.data_analytics.superstore
GROUP BY `Product Name`
)

SELECT Product_Segment,
COUNT(*) AS No_of_Products
FROM product_segment
GROUP BY Product_Segment
ORDER BY No_of_Products DESC;

Product_Segment,No_of_Products
Low Performer,1642
Average Performer,163
High Performer,30
Best Seller,14


📌 Module 8 — Executive Insights

After every query

write

Finding

↓

Business Insight

↓

Recommendation

## Finding 1

Technology products generate the highest revenue among all categories.

Business Insight

Technology products have strong customer demand and contribute significantly to overall sales.

Recommendation

Increase inventory and marketing investment in Technology products while monitoring stock availability.

## Finding 2

Some products generate high revenue but low profit margins.

Business Insight

High discounts or procurement costs may be reducing profitability despite strong sales.

Recommendation

Review pricing strategy and supplier costs for these products.

## Finding 3

Several products consistently generate negative profits.

Business Insight

These products reduce overall company profitability.

Recommendation

Analyze the root cause, renegotiate supplier contracts, revise pricing, or discontinue products with persistent losses.

## Finding 4

A small number of products contribute a large share of total revenue.

Business Insight

Revenue is concentrated in a limited set of products, making them strategically important.

Recommendation

Ensure consistent stock availability, monitor demand, and prioritize these products in marketing campaigns.